### PR TITLE
docs(examples): add temporal `strftime` examples

### LINKED ISSUE #10354: docs: more temporal expression examples
### Please describe the issue

Would it be worthwhile to add additional examples for temporal expressions? 

I was looking through the API reference docs for [temporal expressions](https://ibis-project.org/reference/expression-temporal) and noticed that many methods do not currently have docstrings. 

One could argue that many of these methods are pretty self-explanatory, but I feel that examples could be helpful for newer users. I put a small example together for `strftime` with what I believe would be common patterns.

``` py
        Examples
        --------
        >>> from datetime import datetime
        >>> import ibis
        >>> ibis.options.interactive = True
        >>> t = ibis.memtable(
        ...     [
        ...         (datetime(2020, 10, 5, 8, 0, 0)),
        ...         (datetime(2020, 11, 10, 10, 2, 15)),
        ...         (datetime(2020, 12, 15, 12, 4, 30)),
        ...     ],
        ...     schema=ibis.Schema({"timestamp_col": "timestamp"}),
        ... )

        Return a string with the year and month.

        >>> t.timestamp_col.strftime("%Y-%m")
        ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
        ┃ Strftime(timestamp_col, '%Y-%m') ┃
        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
        │ string                           │
        ├──────────────────────────────────┤
        │ 2020-10                          │
        │ 2020-11                          │
        │ 2020-12                          │
        └──────────────────────────────────┘
        
        Return a string with the month, day, and year.

        >>> t.timestamp_col.strftime("%B %-d, %Y")
        ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
        ┃ Strftime(timestamp_col, '%B %-d, %Y') ┃
        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
        │ string                                │
        ├───────────────────────────────────────┤
        │ October 5, 2020                       │
        │ November 10, 2020                     │
        │ December 15, 2020                     │
        └───────────────────────────────────────┘
 ```
 
 If people think it's worthwhile, I'd like to start adding more examples here. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by IndexSeek**
## Description of changes

Adding examples for TimestampValue and DateValue `strftime` methods. 

I wanted to cover what I felt like were common usage patterns.

## Issues closed

Resolves #10354 

^^ I intend to keep adding more of these, but don't know it it's worthwhile to keep that issue open.

### CODE REVIEW

**Comment by IndexSeek** [comment]
I was going to add strftime examples for TimeValue, but admittedly, was running into difficulties here.

```pycon
In [1]: from datetime import time

In [2]: from ibis.interactive import *

In [3]: t = ibis.memtable([time(10, 28, 3)], schema=ibis.Schema({"time_col": "time"}))

In [4]: t
Out[4]: 
┏━━━━━━━━━━┓
┃ time_col ┃
┡━━━━━━━━━━┩
│ time     │
├──────────┤
│ 10:28:03 │
└──────────┘

In [5]: t.time_col.strftime("%H:%M")
BinderException: Binder Error: No function matches the given name and argument types 'strftime(TIME, STRING_LITERAL)'. You might need to add explicit type casts.
        Candidate functions:
        strftime(DATE, VARCHAR) -> VARCHAR
        strftime(TIMESTAMP, VARCHAR) -> VARCHAR
        strftime(TIMESTAMP_NS, VARCHAR) -> VARCHAR
        strftime(VARCHAR, DATE) -> VARCHAR
        strftime(VARCHAR, TIMESTAMP) -> VARCHAR
        strftime(VARCHAR, TIMESTAMP_NS) -> VARCHAR
        strftime(TIMESTAMP WITH TIME ZONE, VARCHAR) -> VARCHAR
```

File: ibis/expr/types/temporal.py [hunk 1]
```diff
@@ -339,6 +339,46 @@ def strftime(self, format_str: str) -> ir.StringValue:
...
+        >>> from datetime import date
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable(
+        ...     [
+        ...         date(2020, 10, 5),
+        ...         date(2020, 11, 10),
+        ...         date(2020, 12, 15),
+        ...     ],
+        ...     schema=ibis.Schema({"date_col": "date"}),
```
Comment (cpcloud) [inline ibis/expr/types/temporal.py:354]:
```suggestion
        ...     {"date_col": [
        ...         date(2020, 10, 5),
        ...         date(2020, 11, 10),
        ...         date(2020, 12, 15),
        ...     ]},
```

You'll need to run `ruff` again to format this correctly, if you accept the suggestion.

File: ibis/expr/types/temporal.py [hunk 2]
```diff
@@ -538,6 +578,59 @@ def strftime(self, format_str: str) -> ir.StringValue:
...
+        >>> from datetime import datetime
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable(
+        ...     [
+        ...         (datetime(2020, 10, 5, 8, 0, 0)),
+        ...         (datetime(2020, 11, 10, 10, 2, 15)),
+        ...         (datetime(2020, 12, 15, 12, 4, 30)),
+        ...     ],
+        ...     schema=ibis.Schema({"timestamp_col": "timestamp"}),
```
Comment (cpcloud) [inline ibis/expr/types/temporal.py:593]:
```suggestion
        ...     {"timestamp_col": [
        ...         (datetime(2020, 10, 5, 8, 0, 0)),
        ...         (datetime(2020, 11, 10, 10, 2, 15)),
        ...         (datetime(2020, 12, 15, 12, 4, 30)),
        ...     ]},
```

**Review by cpcloud — APPROVED** [review]
Looks great, just some minor suggestions.

**Comment by IndexSeek** [comment]
> Looks great, just some minor suggestions.

Thank you! Removing the explicit schema does make it more concise. I added those suggestions and linted again.

File: ibis/expr/types/temporal.py [hunk 3]
```diff
@@ -538,6 +579,60 @@ def strftime(self, format_str: str) -> ir.StringValue:
...
+        --------
+        >>> from datetime import datetime
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable(
+        ...     {
+        ...         "timestamp_col": [
+        ...             (datetime(2020, 10, 5, 8, 0, 0)),
+        ...             (datetime(2020, 11, 10, 10, 2, 15)),
+        ...             (datetime(2020, 12, 15, 12, 4, 30)),
```
Comment (cpcloud) [inline ibis/expr/types/temporal.py:593]:
```suggestion
        ...             datetime(2020, 10, 5, 8, 0, 0),
        ...             datetime(2020, 11, 10, 10, 2, 15),
        ...             datetime(2020, 12, 15, 12, 4, 30),
```

**Comment by IndexSeek** [comment]
> The Trino failures are unrelated and already fixed upstream, merging!

Thank you so much! Also, I appreciate the parenthesis fix.

**Pull request merged**